docs/docs/reference/crd-reference/taskdefinition.md

```diff
@@ -396,181 +369,123 @@ This task is then referenced in the
 
-This example fetches the Deno script from a remote webserver at runtime:
+    You can embed python code directly in the task definition.
+    This example prints data stored in the parameters map:
 
-```yaml
-apiVersion: lifecycle.keptn.sh/v1beta1
-kind: KeptnTaskDefinition
-metadata:
-  name: hello-keptn-http
-spec:
-  deno:
-    httpRef:
-      url: "https://www.example.com/yourscript.js"
-```
+    ```yaml
+    {% include "../../assets/crd/python-inline.yaml" %}
+    ```
 
-For another example, see the
-[sample-app](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml).
+### Examples for httpRef script
 
-See the
-[sample-app/version-1](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml)
-PodtatoHead example for a more complete example.
+=== "Example 2: httpRef script for a Deno script"
 
-### Example 3: functionRef for a Deno script
+    This example fetches the Deno script from a remote webserver at runtime:
 
-This example calls another defined task,
-illustrating how one `KeptnTaskDefinition` can build
-on top of other `KeptnTaskDefinition`s.
-In this case, it calls `slack-notification-dev`,
-passing `parameters` and `secureParameters` to that other task:
+    ```yaml
+    {% include "../../assets/crd/examples/httpref-script-for-deno-script.yaml" %}
+    ```
 
-```yaml
-apiVersion: lifecycle.keptn.sh/v1beta1
-kind: KeptnTaskDefinition
-metadata:
-  name: slack-notification-dev
-spec:
-  deno:
-    functionRef:
-      name: slack-notification
-    parameters:
-      map:
-        textMessage: "This is my configuration"
-    secureParameters:
-      secret: slack-token
-```
+    For another example, see the
+    [sample-app](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml).
 
-### Example 4: ConfigMapRef for a Deno script
+    See the
+    [sample-app/version-1](https://github.com/keptn-sandbox/lifecycle-toolkit-examples/blob/main/sample-app/version-1/app-pre-deploy.yaml)
+    PodtatoHead example for a more complete example.
 
-This example references a `ConfigMap` by the name of `dev-configmap`
-that contains the code for the function to be executed.
+=== "Example 2: httpRef for a python-runtime runner"
 
-```yaml
-apiVersion: lifecycle.keptn.sh/v1beta1
-kind: KeptnTaskDefinition
-metadata:
-  name: keptntaskdefinition-sample
-spec:
-  deno:
-    configMapRef:
-      name: dev-configmap
-```
+    You can refer to code stored online.
+    For example, we have a few examples available in the
+    [python-runtime samples](https://github.com/keptn/lifecycle-toolkit/tree/main/runtimes/python-runtime/samples)
+    tree.
 
-### Example 5: ConfigMap for a Deno script
+    Consider the following:
 
-This example illustrates the use of both a `ConfigMapRef` and a `ConfigMap`:
+    ```yaml
+    {% include "../../assets/crd/python-configmap.yaml" %}
```

Comment:
this is under http ref but it is a configmap, can you remove it?

Reply:
Thanks for pointing this out, noticed it's incorrectly referenced in docs.
Should I add it in `ConfigMap for a Python` example ? or just remove it.

Reply:
I think it is already used also here https://keptn--3005.org.readthedocs.build/3005/docs/reference/crd-reference/taskdefinition/#examples-for-configmapref so you can just remove the include for the http ref... 

Yes, this page was a mess because there was so much content! Thanks to this PR now we clearly see what material is missing ❤️  I will make a ticket to fill in all the examples after we merge this!

Reply:
:+1:

Reply:
:heavy_check_mark: done @RealAnna